### DISCUSSION TITLE
undulator module

### DISCUSSION THREAD

**Discussion by rl3418** [svenreiche/Genesis-1.3-Version4#75] [General]
Are the roll off parameters both normalised to k_undulator^2? If that is the case, kx and ky are both very small numbers due to the large gamma value. Why is the default value of ky set to 1?

**Comment by rl3418** [comment]
[matching conditions in linear undulator.pdf](https://github.com/svenreiche/Genesis-1.3-Version4/files/10941108/matching.conditions.in.linear.undulator.pdf)

Seems that the roll-off parameters are the k parameters not the strength parameters (kappa). I guess they are just normalised to k_u. Therefore, ky^2=(2+delta)/2, kx^2=delta/2.

**Comment by rl3418** [comment]
One thing that still puzzles me a bit is that the manual says the roll-off parameters are normalised to k_u^2. Is that a typo or the roll-off parameters here define kx^2 and ky^2?

**Comment by svenreiche** [comment]
The effective total focusing strength of an undulator is  f = k_U^2*K^2/gamma^2  but one can distribute them between the planes by shaping the pole faces.

So in x the effective focusing strength is r*f and for y (1-r)*f.
In Genesis terms we have kx = r and ky = (1-r) and has the default values of

kx=0 and ky=1 for a planar undulator (this corresponds the case with infinite wide pole faces) and
kx=0.5 and ky = 0.5 for helical undulator.

In cases of exotic undulators, such as Apple-II there parameters can be large, e.g. kx = 5.2, ky = -4.2
but as long as kx+ky = 1 there is a possible physical implementation of the field properties.

Genesis does not check whether the condition is fulfilled, so one can do unphysical things like kx = 10 and ky = 10.  This makes sense if you want to simulate a constant focusing channel but do not explicit set-up a superimposed FODO lattice.

Hope this helps
Sven

-----------------------------------
Paul Scherrer Institut
Sven Reiche
WBBA/315
CH-5232 Villigen PSI

Telefon: +41 56 310 51 19
E-Mail: ***@***.******@***.***>***@***.***>***@***.***>

________________________________

**Comment by rl3418** [comment]
Thanks for the clarification. What is the easiest way to produce a plot of the evolution of the twiss parameters through the lattice? Do I have to dump the beam profiles and do postprocessing?

**Comment by svenreiche** [comment]
The twiss parameters are not plotted as a function along z but one can easily get them from the rms beamsizes, since the integrator of the transverse coordinate is symplectic, meaning that the emittance is preserved along the beamline and thus can be taken as a constant. If one set to a very small current it is also the same for the energy thus the variation in the beamsize is solely given by the beta-functions


Best
Sven

-----------------------------------
Paul Scherrer Institut
Sven Reiche
WBBA/315
CH-5232 Villigen PSI

Telefon: +41 56 310 51 19
E-Mail: ***@***.******@***.***>***@***.***>***@***.***>

________________________________
